src/overlays/actors/ovl_Obj_Snowball2/z_obj_snowball2.c

```diff
@@ -28,82 +38,589 @@ const ActorInit Obj_Snowball2_InitVars = {
+                                          this->actor.world.pos.z, &this->actor.shape.rot);
+    Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+    Collider_UpdateSpheres(0, &this->collider);
+}
+
+void func_80B38E88(ObjSnowball2* this, GlobalContext* globalCtx) {
+    s32 temp_v0;
+
+    if (this->unk_1AE == 0) {
+        temp_v0 = func_800A8150(ENOBJSNOWBALL2_GET_3F(&this->actor));
+        if (temp_v0 >= 0) {
+            Item_DropCollectible(globalCtx, &this->actor.world.pos,
+                                 (ENOBJSNOWBALL2_GET_7F00(&this->actor) << 8) | temp_v0);
+            this->unk_1AE = 1;
+        }
+    }
+}
+
+void func_80B38EFC(ObjSnowball2* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 i;
+    Vec3f spA4;
+    Vec3f sp98;
+    Vec3f sp8C;
+    f32 sp88 = this->actor.world.pos.y + (this->actor.shape.yOffset * this->actor.scale.y);
+
+    for (i = 0; i < 2; i++) {
+        sp8C.x = ((Rand_ZeroOne() * 30.0f) - 15.0f) + this->actor.world.pos.x;
+        sp8C.y = ((Rand_ZeroOne() * 20.0f) - 10.0f) + sp88;
+        sp8C.z = ((Rand_ZeroOne() * 30.0f) - 15.0f) + this->actor.world.pos.z;
+
+        sp98.x = (Rand_ZeroOne() - 0.5f) * 1.6f;
+        sp98.y = -1.5f;
+        sp98.z = (Rand_ZeroOne() - 0.5f) * 1.6f;
+
+        spA4.x = sp98.x * -0.06f;
+        spA4.y = sp98.y * -0.06f;
+        spA4.z = sp98.z * -0.06f;
+
+        func_800B0E48(globalCtx, &sp8C, &sp98, &spA4, &D_80B3A914, &D_80B3A918, (s32)(Rand_ZeroOne() * 40.0f) + 10,
+                      (s32)(Rand_ZeroOne() * 30.0f) + 10);
+    }
+}
+
+void func_80B39108(ObjSnowball2* this, GlobalContext* globalCtx) {
+    s32 phi_s5;
+    s16 phi_s1;
+    s32 phi_v0;
+    s32 i;
+    u32 temp;
+    s32 temp_s3;
+    Vec3f spDC;
+    Vec3f spD0;
+    f32 temp_f2;
+    f32 spC8 = this->actor.world.pos.y + (this->actor.shape.yOffset * this->actor.scale.y);
+    f32 temp_f0;
+    f32 temp_f12;
+    f32 temp_f20;
+    f32 temp_f22;
+    f32 temp_f24;
+
+    for (i = 0, phi_s5 = 0; i < 11; i++, phi_s5 += 0x1745) {
+        temp_s3 = (s32)(Rand_ZeroOne() * 5957.0f) + phi_s5;
+
+        if (((u32)Rand_Next() >> 0x1E) == 0) {
+            phi_s1 = 0x20;
+        } else {
+            phi_s1 = 0x40;
+        }
+
+        if ((i & 3) == 0) {
+            phi_s1 |= 1;
+        }
+
+        temp_f0 = Rand_ZeroOne();
+        temp_f20 = (1.0f - SQ(temp_f0)) * 12.0f;
+        temp_f22 = Math_SinS(temp_s3);
+        temp_f24 = Math_CosS(temp_s3);
+        temp_f12 = (Rand_ZeroOne() * 1.6f) - 0.8f;
+        temp_f2 = fabsf(temp_f12) * temp_f12;
+
+        spDC.x = temp_f22 * 4.0f;
+        spDC.y = (temp_f2 * 6.0f) + 9.0f;
+        spDC.z = temp_f24 * 4.0f;
+
+        spD0.x = (temp_f22 * temp_f20) + this->actor.world.pos.x;
+        spD0.y = (temp_f2 * temp_f20) + spC8;
+        spD0.z = (temp_f24 * temp_f20) + this->actor.world.pos.z;
+        temp = Rand_Next();
+
+        if ((i & 3) == 0) {
```

Comment:
Would this match with `%4` instead? If not, could you change it to `!(i & 3)`?

Reply:
Why `!`? it's not a bool.